napari/_qt/threading.py

```diff
@@ -0,0 +1,840 @@
+    return QThreadPool.globalInstance().activeThreadCount()
+
+
+#############################################################################
+
+# convenience functions for creating Worker instances
+
+
+def create_worker(
+    func: Callable,
+    *args,
+    _start_thread: Optional[bool] = None,
+    _connect: Optional[Dict[str, Union[Callable, Sequence[Callable]]]] = None,
+    _worker_class: Optional[Type[WorkerBase]] = None,
+    _ignore_errors: bool = False,
+    **kwargs,
+) -> WorkerBase:
+    """Convenience function to start a function in another thread.
+
+    By default, uses :class:`Worker`, but a custom ``WorkerBase`` subclass may
+    be provided.  If so, it must be a subclass of :class:`Worker`, which
+    defines a standard set of signals and a run method.
+
+    Parameters
+    ----------
+    func : Callable
+        The function to call in another thread.
+    _start_thread : bool, optional
+        Whether to immediaetly start the thread.  If False, the returned worker
+        must be manually started with ``worker.start()``. by default it will be
+        ``False`` if the ``_connect`` argument is ``None``, otherwise ``True``.
+    _connect : Dict[str, Union[Callable, Sequence]], optional
+        A mapping of ``"signal_name"`` -> ``callable`` or list of ``callable``:
+        callback functions to connect to the various signals offered by the
+        worker class. by default None
+    _worker_class : Type[WorkerBase], optional
+        The :class`WorkerBase` to instantiate, by default
+        :class:`FunctionWorker` will be used if ``func`` is a regular function,
+        and :class:`GeneratorWorker` will be used if it is a generator.
+    _ignore_errors : bool, optional
+        If ``False`` (the default), errors raised in the other thread will be
+        reraised in the main thread (makes debugging significantly easier).
+    *args
+        will be passed to ``func``
+    **kwargs
+        will be passed to ``func``
+
+    Example
+    -------
+
+    .. code-block:: python
+
+        def long_function(duration):
+            import time
+            time.sleep(duration)
+
+        worker = create_worker(long_function, 10)
+
+    Returns
+    -------
+    WorkerBase
+        An instantiated worker.  If ``_start_thread`` was ``False``, the worker
+        will have a `.start()` method that can be used to start the thread.
+
+    Raises
+    ------
+    TypeError
+        If a worker_class is provided that is not a subclass of WorkerBase.
+    TypeError
+        If _connect is provided and is not a dict of ``{str: callable}``
+    """
+    if not _worker_class:
+        if inspect.isgeneratorfunction(func):
+            _worker_class = GeneratorWorker
+        else:
+            _worker_class = FunctionWorker
+
+    if inspect.isclass(_worker_class) and issubclass(
+        WorkerBase, _worker_class
+    ):
+        raise TypeError(f'Worker {_worker_class} must be a subclass of Worker')
```

Comment:
should this be **not** issubclass???

Reply:
I'm very confused about this clause...

Reply:
it also seems like the classes are inverted. ie I would write `and not issubclass(_worker_class, WorkerBase)`. But I'm not sure whether that's the intent or I'm misunderstanding something.

Reply:
yeah, that was all wrong.  thanks